application/app/src/main/java/uk/nhs/digital/uec/api/controller/FuzzyServiceSearchController.java

```diff
@@ -39,89 +44,93 @@
+            @ApiParam(POSTCODE_PRIORITY_DESC) @RequestParam(name = "postcode_priority", required = false) Integer postcodePriority,
+            @ApiParam(PUBLIC_NAME_PRIORITY_DESC) @RequestParam(name = "public_name_priority", required = false) Integer publicNamePriority)
+            throws NotFoundException, InvalidParameterException {
+        log.info("Incoming request param - postcode: {}", searchPostcode);
+        log.info("Incoming request param - search_term: {}", searchCriteria);
+        log.info("Incoming request param - search_latitude: {}, search_longitude:{}", searchLatitude,searchLongitude);
+        utils.configureApiRequestParams(
+                fuzzLevel,
+                null, // filterReferralRole will be implemented in future
+                maxNumServicesToReturnFromEs,
+                maxNumServicesToReturn,
+                namePriority,
+                addressPriority,
+                postcodePriority,
+                publicNamePriority);
 
-    final ApiSuccessResponse response = new ApiSuccessResponse();
-    final ApiSearchResultsResponse searchResultsResponse = new ApiSearchResultsResponse();
+        final ApiSearchParamsResponse searchParamsResponse = new ApiSearchParamsResponse.ApiSearchParamsResponseBuilder()
+                .searchCriteria(searchCriteria == null || searchCriteria.isEmpty()
+                        ? List.of(searchPostcode.substring(0, 4).trim())
+                        : searchCriteria)
+                .searchPostcode(searchPostcode)
+                .searchLatitude(searchLatitude)
+                .searchLongitude(searchLongitude)
+                .fuzzLevel(requestParams.getFuzzLevel())
+                .addressPriority(requestParams.getAddressPriority())
+                .postcodePriority(requestParams.getPostcodePriority())
+                .publicNamePriority(requestParams.getPublicNamePriority())
+                .namePriority(requestParams.getNamePriority())
+                .maxNumServicesToReturn(requestParams.getMaxNumServicesToReturn())
+                .build();
 
-    final List<DosService> dosServices =
-        fuzzyServiceSearchService.retrieveServicesByFuzzySearch(searchPostcode, searchCriteria);
-    searchResultsResponse.setServices(dosServices);
-    response.setSearchParameters(searchParamsResponse);
-    response.setSearchResults(searchResultsResponse);
+        final ApiSuccessResponse response = new ApiSuccessResponse();
+        final ApiSearchResultsResponse searchResultsResponse = new ApiSearchResultsResponse();
+        final boolean areValuesNumbers = NumberUtils.isCreatable(searchLatitude) && NumberUtils.isCreatable(searchLongitude);
+        log.info("Is lat and lng values are numbers: {}", areValuesNumbers);
+        final List<DosService> dosServices =  areValuesNumbers ?
+        fuzzyServiceSearchService.retrieveServicesByGeoLocation(searchLatitude, searchLongitude, searchCriteria):
```

Comment:
Use one method and pass in all the variables. Then in the service implementation we will determine which search to use based on which variables are provided. The Lat and Long search should take priority because on initial load of the fuzzy services the user hasn't typed in any search terms